metadata/metadata_test.go

```diff
@@ -69,3 +71,50 @@ func TestJoin(t *testing.T) {
 		}
 	}
 }
+
+func TestAppendToOutgoingContext(t *testing.T) {
+	for _, test := range []struct {
+		in   []string
+		want MD
+	}{
+		{[]string{"k1", "v1"}, MD{"k1": []string{"v1"}}},
+		{[]string{"k1", "v1", "k1", "v2", "k2", "v3"}, MD{"k1": []string{"v1", "v2"}, "k2": []string{"v3"}}},
+	} {
+		ctx := context.Background()
+		ctx = AppendToOutgoingContext(ctx, test.in...)
```

Comment:
These test cases don't cover the actual behavior of appending; you could replace this with `NewOutgoingContext` and the tests would still pass; please append to a context that was created with `NewOutgoingContext`, and also do more than one append call to a single context to make sure all of that is working as expected.

Reply:
Done. One test with no existing MD, and one test with existing MD + two appends